packages/hawtio/src/plugins/connect/init.ts

```diff
@@ -24,25 +25,35 @@ export async function isConnectionStatusActive(): Promise<boolean> {
   return connectService.getCurrentConnectionId() !== null
 }
 
+let proxyEnabled: boolean | null = null
+
 async function isProxyEnabled(): Promise<boolean> {
+  if (proxyEnabled != null) {
+    return proxyEnabled
+  }
```

Comment:
Be mindful that even in the single threaded model this guard doesn't perfectly eliminate the possibility of the following code being executed twice, as there are `await`s before setting the variable in the following.